src/components/AuditLogsTable/AuditLogsTablePagination/AuditLogsTablePagination.test.tsx

```diff
@@ -0,0 +1,82 @@
+import { screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { add } from "date-fns";
+
+import type { RootState } from "store/store";
+import { rootStateFactory, jujuStateFactory } from "testing/factories";
+import { generalStateFactory, configFactory } from "testing/factories/general";
+import { auditEventFactory } from "testing/factories/juju/jimm";
+import { auditEventsStateFactory } from "testing/factories/juju/juju";
+import { renderComponent } from "testing/utils";
+
+import AuditLogsTablePagination, { Label } from "./AuditLogsTablePagination";
+
+describe("AuditLogsTablePagination", () => {
+  let state: RootState;
+
+  beforeEach(() => {
+    state = rootStateFactory.build({
+      general: generalStateFactory.build({
+        config: configFactory.build({
+          controllerAPIEndpoint: "wss://example.com/api",
+        }),
+        controllerConnections: {
+          "wss://example.com/api": {
+            user: {
+              "display-name": "eggman",
+              identity: "user-eggman@external",
+              "controller-access": "",
+              "model-access": "",
+            },
+          },
+        },
+      }),
+      juju: jujuStateFactory.build({
+        auditEvents: auditEventsStateFactory.build({
+          items: [
+            auditEventFactory.build({
+              time: add(new Date(), { days: -1 }).toISOString(),
+            }),
+          ],
+          loaded: true,
+        }),
+      }),
+    });
+  });
+
+  it("should navigate to next page and then to previous page", async () => {
+    state.juju.auditEvents.items = auditEventFactory.buildList(51);
```

Comment:
I remembered that the factories have `buildList()` instead of having to create them in a loop.